src/compiler/crystal/codegen/link.cr

```diff
@@ -105,7 +105,8 @@ module Crystal
     end
 
     private def lib_flags_posix
-      library_path = ["/usr/lib", "/usr/local/lib"]
+      library_path = ENV["LIBRARY_PATH"]?.try(&.split(':', remove_empty: true)) ||
+                     ["/usr/lib", "/usr/local/lib"]
```

Comment:
Equivalent change, to make the absurdity explicit.

```suggestion
      library_path = ENV.has_key?("LIBRARY_PATH") ? [] of String :
                     ["/usr/lib", "/usr/local/lib"]
```

Reply:
To be fair, there's also nothing wrong with what this code accomplished, it just has a potential to be misunderstood / cargo-culted.

Reply:
But I do want to look up the static libraries in the LIBRARY_PATH directories.

Reply:
That is already the case

Reply:
I think what oprypin means is that we don't have to search them ourselves because the linker will do that. So maybe all of this is dead code? I'm not sure.

Reply:
That said, LIBRARY_CRYSTAL is empty for me and without this it will stop working.

Reply:
```suggestion
      # Note: most linkers (Linux, macOS) already look at `LIBRARY_PATH`
      # themselves (and also have these defaults), so this code would only
      # duplicate that work, but the duplication is harmless.
      library_path = ENV["LIBRARY_PATH"]?.try(&.split(':', remove_empty: true)) ||
                     ["/usr/lib", "/usr/local/lib"]
```

Reply:
I still hold that the behavior changes. Without library_path being set `-lfoo` will be passed to the linker, instead of `/path/to/libfoo.a`. This comes from the `find_static_lib` invocation.